shared/docs/flows/AppHash.md

```diff
@@ -1,40 +1,92 @@
-# AppHash
+# AppHash <!-- omit in toc -->
+
+- [Context Initialization](#context-initialization)
+- [Block Application](#block-application)
+- [Block Commit](#block-commit)
 
 ## Context Initialization
 
+This flow shows the process of context initialization between all the modules requirer to apply a block and compute a state hash during the consensus lifecycle.
+
 ```mermaid
 sequenceDiagram
     %% autonumber
     participant N as Node
     participant C as Consensus
     participant U as Utility
     participant P as Persistence
-    participant PP as Persistence (PostgresDB)
-    participant PM as Persistence (MerkleTree)
     participant P2P as P2P
 
-    %% Should this be P2P?
     N-->>C: HandleMessage(anypb.Any)
     critical NewRound Message
         C->>+U: NewContext(height)
         U->>P: NewRWContext(height)
         P->>U: PersistenceRWContext
-        U->>U: Store persistenceContext
+        U->>U: store context<br>locally
         U->>-C: UtilityContext
-        C->>C: Store utilityContext
-        Note over C, PM: See 'Block Application'
+        C->>C: store context<br>locally
+        Note over C, P: See 'Block Application'
     end
+
     Note over N, P2P: Hotstuff lifecycle
     N-->>C: HandleMessage(anypb.Any)
+
     critical Decide Message
-        Note over C, PM: See 'Block Commit'
+        Note over C, P: See 'Block Commit'
     end
 ```
 
 ## Block Application
 
-TODO(olshansky): Add a sequenceDiagram here.
+```mermaid
+sequenceDiagram
+    participant C as Consensus
+    participant U as Utility
+    participant P as Persistence
+
+    alt as leader
+        C->>+U: GetProposalTransactions(proposer, maxTxBz, [lastVal])
+        U->>U: reap mempool
```

Comment:
From this diagram it looks like if the `leader` reaps the mempool **every time**.

But as far as I understand, it has to only if there is no quorum for the `Prepare` otherwise it has to use the block in the QC.

I understand it would be quite challenging to represent the two scenarios but maybe it could be another  `alt`

Reply:
What if I have a `Note` with the text `Get / Prepare Block` and in a separate diagram, we can show the difference?

Reply:
Resolving per the discussion below.